Mathlib/Tactic/ReduceModChar.lean

```diff
@@ -308,21 +289,12 @@ syntax (name := reduce_mod_char!) "reduce_mod_char!" (location)? : tactic
 
 elab_rules : tactic
 | `(tactic| reduce_mod_char $[$loc]?) => unsafe do
-  match expandOptLocation (Lean.mkOptionalNode loc) with
-  | Location.targets hyps target => do
-    (← getFVarIds hyps).forM reduceModCharHyp
-    if target then reduceModCharTarget
-  | Location.wildcard => do
-    (← (← getMainGoal).getNondepPropHyps).forM reduceModCharHyp
-    reduceModCharTarget
+  let loc := (loc.map expandLocation).getD (.targets #[] true)
```

Comment:
I am relatively sure this is fine, but have not verified that switching from `getNondepPropHyps` to `expandLocaltion` is precisely behaviour-preserving.

Reply:
Good catch!

I abstracted what was being done here (filtering the NondepProp hypotheses) as a variant of `Elab.Tactic.withLocation`, which I have named `Elab.Tactic.withNondepPropLocation`. I changed `reduce_mod_char` and `norm_num` to use this, and I added Arend's test.